### ISSUE TITLE
Stack doesn't build

### ISSUE THREAD

**Issue by kolmodin** [commercialhaskell/stack#3134]
Tried to build stack with cabal;

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.2

Configuring stack-0.1.2.0...
Preprocessing library stack-0.1.2.0...
[ 1 of 58] Compiling System.Process.PagerEditor ( src/System/Process/PagerEditor.hs, dist/dist-sandbox-3ae950eb/build/System/Process/PagerEditor.o )
[ 2 of 58] Compiling System.Process.Log ( src/System/Process/Log.hs, dist/dist-sandbox-3ae950eb/build/System/Process/Log.o )
[ 3 of 58] Compiling System.Process.Read ( src/System/Process/Read.hs, dist/dist-sandbox-3ae950eb/build/System/Process/Read.o )
[ 4 of 58] Compiling System.Process.Run ( src/System/Process/Run.hs, dist/dist-sandbox-3ae950eb/build/System/Process/Run.o )
[ 5 of 58] Compiling Stack.New        ( src/Stack/New.hs, dist/dist-sandbox-3ae950eb/build/Stack/New.o )
[ 6 of 58] Compiling Stack.FileWatch  ( src/Stack/FileWatch.hs, dist/dist-sandbox-3ae950eb/build/Stack/FileWatch.o )
[ 7 of 58] Compiling Paths_stack      ( dist/dist-sandbox-3ae950eb/build/autogen/Paths_stack.hs, dist/dist-sandbox-3ae950eb/build/Paths_stack.o )
[ 8 of 58] Compiling Path.IO          ( src/Path/IO.hs, dist/dist-sandbox-3ae950eb/build/Path/IO.o )

src/Path/IO.hs:227:33: error:
    Ambiguous occurrence ‘listDirectory’
    It could refer to either ‘System.Directory.listDirectory’,
                             imported from ‘System.Directory’ at src/Path/IO.hs:34:1-33
                          or ‘Path.IO.listDirectory’, defined at src/Path/IO.hs:107:1
cabal: Leaving directory '/tmp/cabal-tmp-20652/stack-0.1.2.0'
cabal: Error: some packages failed to install:
stack-0.1.2.0-6TYerA924671A2u8fGONAA failed during the building phase. The
exception was:
ExitFailure 1
```

**Comment by mgsloan** [comment]
Hmm, that's quite an old version of stack!  Not sure why it'd get chosen.

**Comment by kolmodin** [comment]
Hmm, indeed! I tried to install stack into a non-empty sandbox. Probably had some old version of some package that interfered.
It resolved to stack 1.4.0 when I tried with a fresh sandbox. Sorry for the noise!
